src/js/productData.mjs

```
@@ -14,5 +14,8 @@ export function getData(category = 'tents') {
 
 export async function findProductById(id) {
   const products = await getData();
+  if (products === undefined) {
+    throw new Error(`Product with id ${id} not found.`);
+  }
```

we should throw the error here instead of in the other try block.